privly.safariextension/scripts/content_scripts/privly.js

```diff
@@ -395,6 +400,14 @@ var privly = {
 
     //put the iframe into the page
     object.parentNode.insertBefore(iFrame, object);
+
+    // Give the iframe 2s to load. If CSP blocks iframe injection then,
+    // stop the content script.
+    setTimeout(function() {
+      if (privly.blockedByCSP === true) {
+        privly.stop();
+      }
+    }, 2000);
```

Comment:
A better approach may be to define an html document in the Safari extension, `CSP_iframe.html`, whose only contents are `<html><head></head><body></body></html>`. You could then start injecting the links as soon as the CSP iframe loads. What do you think?